DSCResources/MSFT_xVMSwitch/MSFT_xVMSwitch.psm1

```diff
@@ -66,21 +76,41 @@ function Set-TargetResource
         # If switch is present and it is external type, that means it doesn't have right properties (TEST code ensures that)
         if($switch -and ($switch.SwitchType -eq 'External'))
         {
-            Write-Verbose -Message "Checking switch $Name NetAdapterInterface ..."
+            $removeReaddSwitch = $false
+
+            Write-Verbose -Message "Checking switch $Name NetAdapterInterface and BandwidthReservationMode ..."
             if((Get-NetAdapter -Name $NetAdapterName).InterfaceDescription -ne $switch.NetAdapterInterfaceDescription)
             {
-                Write-Verbose -Message "Removing switch $Name and creating with right netadapter ..."
+                Write-Verbose -Message "The switch $Name NetAdapterInterface is incorrect ..."
+                $removeReaddSwitch = $true
+            }
+            elseif(($BandwidthReservationMode -ne "NA") -and ($switch.BandwidthReservationMode -ne $BandwidthReservationMode))
+            {
+                Write-Verbose -Message "The switch $Name BandwidthReservationMode is incorrect ..."
+                $removeReaddSwitch = $true
+            }
+
+            if($removeReaddSwitch)
+            {
+                Write-Verbose -Message "Removing switch $Name and creating with the correct properties ..."
                 $switch | Remove-VMSwitch -Force
                 $parameters = @{}
                 $parameters["Name"] = $Name
                 $parameters["NetAdapterName"] = $NetAdapterName
+                if(($BandwidthReservationMode -ne "NA") -and ((Get-WmiObject -Class 'Win32_OperatingSystem').Version -ge 6.2.0))
```

Comment:
If the parameter is specified and the OS does not support it.  There should be an error because we are not in the desired state.